### PR TITLE
User improvements

### PR DESCRIPTION
**Pull request by cremebrule**

### CODE REVIEW

File: omnigibson/robots/robot_base.py [hunk 1]
```diff
@@ -189,6 +194,9 @@ def _initialize(self):
         for sensor in self._sensors.values():
             sensor.initialize()
 
+        # Load the observation space for this robot
+        self.load_observation_space()
+
```
Comment (ChengshuLi) [inline omnigibson/robots/robot_base.py:199]: Interesting, did we not have this before?

File: omnigibson/simulator.py [hunk 2]
```diff
@@ -474,8 +475,9 @@ def _transition_rule_step(self):
         for added_obj_attr in added_obj_attrs:
             new_obj = added_obj_attr.obj
             self.import_object(added_obj_attr.obj)
-            pos, orn = added_obj_attr.pos, added_obj_attr.orn
-            new_obj.set_position_orientation(position=pos, orientation=orn)
+            new_obj.set_position_orientation(position=added_obj_attr.pos, orientation=added_obj_attr.orn)
+            if isinstance(new_obj, DatasetObject):
+                new_obj.set_bbox_center_position_orientation(position=added_obj_attr.bb_pos, orientation=added_obj_attr.bb_orn)
```
Comment (ChengshuLi) [inline omnigibson/simulator.py:480]: This seems a bit too hacky. Is all dataset object using the set_bbx_center_position_orientation mechanism?

File: omnigibson/simulator.py [hunk 3]
```diff
@@ -489,8 +491,9 @@ def play(self):
             # Track whether we're starting the simulator fresh -- i.e.: whether we were stopped previously
             was_stopped = self.is_stopped()
 
-            # Minimize physics leakage by slowing down simulator significantly
-            with self.slowed(dt=1e-3):
+            # Run super first
+            # We suppress warnings from omni.usd because it complains about values set in the native USD
```
Comment (ChengshuLi) [inline omnigibson/simulator.py:495]: sorry what does this mean?

File: omnigibson/simulator.py [hunk 4]
```diff
@@ -505,9 +508,12 @@ def play(self):
...
                 self._objects_to_initialize = []
-                # Also update the scene registry
-                # TODO: A better place to put this perhaps?
-                self._scene.object_registry.update(keys="root_handle")
+
+            # If we were stopped, take an additional sim step to make sure simulator is functioning properly
+            # We need to do this because for some reason omniverse exhibits strange behavior if we do certain operations
+            # immediately after playing; e.g.: syncing USD poses when flatcache is enabled
+            if was_stopped:
+                self.step()
```
Comment (ChengshuLi) [inline omnigibson/simulator.py:516]: hmm another physics leakage. Is it absolutely needed? would self.render() suffice here?

File: omnigibson/tasks/dummy_task.py [hunk 5]
```diff
@@ -26,7 +26,6 @@ def _create_reward_functions(self):
     def _reset_agent(self, env):
         # Place agent(s) at origin by default
         for robot in env.robots:
-            robot.reset()
```
Comment (ChengshuLi) [inline omnigibson/tasks/dummy_task.py:29]: Is this because land_object has robot.reset() in it?

File: omnigibson/transition_rules.py [hunk 6]
```diff
@@ -277,37 +278,42 @@ def transition(self, individual_objects, group_objects):
...
+
+            # Get bounding box info
+            part_bb_pos = np.array(part["bb_pos"])
+            part_bb_orn = np.array(part["bb_orn"])
+
+            # Determine the relative scale to apply to the object part from the original object
+            # Note that proper (rotated) scaling can only be applied when the relative orientation of
+            # the object part is a multiple of 90 degrees wrt the parent object, so we assert that here
+            # Check by making sure the real component of rotation is close to 0, 0.707, or 1
+            assert np.any(np.isclose(np.ones(3) * part_bb_orn[-1], np.array([1.0, 0.7071, 0.0]), atol=1e-3)), \
```
Comment (ChengshuLi) [inline omnigibson/transition_rules.py:292]: hmm I am not sure this is the best way to compare two orientations. Can we use a more principled method?

File: omnigibson/transition_rules.py [hunk 7]
```diff
@@ -277,37 +278,42 @@ def transition(self, individual_objects, group_objects):
...
                 prim_path=f"/World/{part_obj_name}",
-                usd_path=usd_path,
-                category=part_category,
                 name=part_obj_name,
-                scale=sliced_obj.scale,
-                abilities={}
+                category=part["category"],
+                model=part["model"],
+                bounding_box=part["bb_size"] * scale,
+                abilities={},
```
Comment (ChengshuLi) [inline omnigibson/transition_rules.py:312]: actually it doesn't seem right to pass in abilities={} here.

File: omnigibson/controllers/ik_controller.py [hunk 8]
```diff
@@ -5,6 +5,10 @@
 from omnigibson.utils.processing_utils import MovingAverageFilter
 from omnigibson.utils.control_utils import IKSolver
 from omnigibson.utils.python_utils import assert_valid_key
+from omnigibson.utils.ui_utils import create_module_logger
+
+# Create module logger
+log = create_module_logger(module_name=__name__)
```
[the remark this answers is not in the document]
Comment (ChengshuLi) [reply omnigibson/controllers/ik_controller.py:11]: It help tell you where the logging message is from (which module).

File: omnigibson/simulator.py (hunk 2 above)
Comment (ChengshuLi) [reply omnigibson/simulator.py:480]: To a bit of sanity check here to safeguard added_obj_attr.attr exists.

Comment (ChengshuLi) [reply omnigibson/simulator.py:480]: it's a either/or situation, not an overwritten situation.

File: omnigibson/simulator.py (hunk 3 above)
Comment (ChengshuLi) [reply omnigibson/simulator.py:495]: maybe add a bit more explicit comment here.

File: omnigibson/simulator.py (hunk 4 above)
Comment (ChengshuLi) [reply omnigibson/simulator.py:516]: no

File: omnigibson/transition_rules.py (hunk 6 above)
Comment (ChengshuLi) [reply omnigibson/transition_rules.py:292]: do you black magic

File: omnigibson/transition_rules.py [hunk 9]
```diff
@@ -277,37 +278,42 @@ def transition(self, individual_objects, group_objects):
...
             part_obj = DatasetObject(
                 prim_path=f"/World/{part_obj_name}",
-                usd_path=usd_path,
-                category=part_category,
                 name=part_obj_name,
-                scale=sliced_obj.scale,
-                abilities={}
+                category=part["category"],
+                model=part["model"],
+                bounding_box=part["bb_size"] * scale,
```
Comment (ChengshuLi) [inline omnigibson/transition_rules.py:311]: Maybe add a bit more comments regarding what the scale of the child object should be, i.e. it should actually be `part["bb_size"] * scale / self.native_bbox`.

**Review by ChengshuLi — APPROVED** [review]

File: omnigibson/simulator.py (hunk 3 above)
Comment (cremebrule) [reply omnigibson/simulator.py:496]: done!

File: omnigibson/simulator.py (hunk 2 above)
Comment (cremebrule) [reply omnigibson/simulator.py:481]: done!

File: omnigibson/transition_rules.py (hunk 6 above)
Comment (cremebrule) [reply omnigibson/transition_rules.py:292]: black magic done

File: omnigibson/transition_rules.py (hunk 7 above)
Comment (cremebrule) [reply omnigibson/transition_rules.py:312]: done!

File: omnigibson/transition_rules.py (hunk 9 above)
Comment (cremebrule) [reply omnigibson/transition_rules.py:311]: done!

**Pull request merged**